common/models/model-method.js

```diff
@@ -10,4 +11,35 @@ module.exports = function(ModelMethod) {
    * @class ModelMethod
    * @inherits WorkspaceEntity
    */
-};
+
+  ModelMethod._shouldEncodeStaticFlagInName = function() {
+    var version = ModelMethod.app.models.Workspace.loopBackVersion;
+    return version != null ? !semver.gtr('3.0.0', version) : false;
+  };
+
+  ModelMethod.getJsonKey = function(data) {
+    if (!this._shouldEncodeStaticFlagInName()) {
+      return data.name;
+    }
+    var isStatic = data.isStatic;
+    if (isStatic !== undefined) {
+      var name = data.name;
+      var matchName = name.match(/^prototype\.(.*)$/);
+      if (!isStatic && (matchName === null || !matchName)) {
+        data.name = 'prototype.' + name;
+      }
+    }
+    delete data.isStatic;
```

Comment:
Oh, `getConfigFromData` already contains code to remove `config.isStatic`. I think it should be enough to delete this line?